python/ray/data/_internal/plan.py

```diff
@@ -136,6 +136,11 @@ def __init__(
         # determined by the config at the time it was created.
         self._context = copy.deepcopy(DataContext.get_current())
 
+        # Whether the corresponding dataset is generated from a pipeline.
+        # Currently, when this is True, this skips the new execution plan optimizer.
+        # TODO(scottjlee): remove this once we remove DatasetPipeline.
+        self._generated_from_pipeline = not self._context.optimizer_enabled
```

Comment:
This should be true regardless of the optimizer flag. Because in `_get_execution_dag`, we will check the optimizer flag.